prow/git/v2/interactor.go

```diff
@@ -45,7 +46,7 @@ type Interactor interface {
 	// Merge merges the commitlike into the current HEAD
 	Merge(commitlike string) (bool, error)
 	// MergeWithStrategy merges the commitlike into the current HEAD with the strategy
-	MergeWithStrategy(commitlike, mergeStrategy string) (bool, error)
+	MergeWithStrategy(commitlike, mergeStrategy string, opts ...git.MergeOpt) (bool, error)
```

Comment:
These should be separate types

Reply:
Since the opts are optional, I don't think that it will affect anything. Do you think that its better to have a `MergeWithStrategyAndOptions` type here?

Reply:
Yes. v2 should not import v1.